sphinx/directives/other.py

```diff
@@ -39,6 +39,7 @@ class TocTree(Directive):
     final_argument_whitespace = False
     option_spec = {
         'maxdepth': int,
+        'name': str,
```

Comment:
Agreed, backward compatibility is necessary. Alternatively, I propose `:caption:` instead of `:name:`.